features/roomlist/impl/src/main/kotlin/io/element/android/features/roomlist/impl/datasource/RoomListDataSource.kt

```diff
@@ -92,6 +102,16 @@ class RoomListDataSource @Inject constructor(
     val allRooms: StateFlow<ImmutableList<RoomListRoomSummary>> = _allRooms
     val filteredRooms: StateFlow<ImmutableList<RoomListRoomSummary>> = _filteredRooms
 
+    @OptIn(FlowPreview::class)
+    private fun observerNotificationSettings() {
+        notificationSettingsService.notificationSettingsChangeFlow
```

Comment:
As per discussions above this is required when we move away from the room list. It's not ideal as it could leak resrouces if we had multiple RoomListDatasources but that shouldn't happen for now. The plan is to remove this code soon when we change the rust sdk to notify us about changes to the room via the room list api(vs this workaround with notificaiton settings api).